gmakefile.test

```diff
@@ -263,19 +263,23 @@ else ifdef searchin
 else ifdef argsearch
   TESTTARGETS := $(foreach v,$(alltesttargets),$(if $(findstring $(argsearch),$($(v)_ARGS)),$(v)))
 else ifdef globsearch
+  # PETSc uses
+  # TESTTARGETS := $(shell $(PYTHON) config/query_tests.py 'name' '$(globsearch)')
+  # since v3.13.1 but it is currently not working for us
   TESTTARGETS := $(shell $(PYTHON) -c"import sys,fnmatch,itertools; m=[fnmatch.filter(sys.argv[2].split(),p) for p in sys.argv[1].split()]; print(' '.join(list(itertools.chain.from_iterable(m))))" '$(globsearch)' '$(alltesttargets)')
 else ifdef test-fail
   TESTTARGETS := $(shell $(PETSC_ARCH)/tests/echofailures.sh)
 else ifdef query
-  TESTTARGETS := $(shell $(PYTHON) config/query_tests.py '$(query)' '$(queryval)')
+  TESTTARGETS := $(shell $(PYTHON) $(PETSCCONFIGDIR)/query_tests.py '$(query)' '$(queryval)')
```

Comment:
```suggestion
  TESTTARGETS := $(shell $(PYTHON) $(PETSCCONFIGDIR)/query_tests.py --testdir=$(TESTDIR) --srcdir=$(TESTSRCDIR) '$(query)' '$(queryval)')
```
otherwise it fails